validphys2/src/validphys/closuretest/multiclosure_nsigma_helpers.py

```diff
@@ -0,0 +1,170 @@
+"""
+This module contains some helper functions that are used for the computation of nsigma in the
+context of a multi-closure test.
+"""
+
+import dataclasses
+import pandas as pd
+import numpy as np
+import logging
+
+from reportengine import collect
+
+from validphys.core import DataSetSpec, PDF
+from nnpdf_data.coredata import CommonData
+from validphys.calcutils import calc_chi2
+from validphys import convolution
+
+log = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class CentralChi2Data:
+    value: float
+    ndata: int
+    dataset: DataSetSpec
+
+    @property
+    def reduced(self):
+        return self.value / self.ndata
```

Comment:
There's already `Chi2Data` in results... I wonder whether it'd be better to extend that one? Or maybe use a different name here?

The difference is that this one holds the dataset only while the other holds a whole result object.

(in any case docstr needed)

Reply:
What name would you go for? Would `DatasetChi2` work for you?

Reply:
Yes. But also, I think it should be in results?

Btw, this is a case (and the one below) where there a clashes with other parts of the code generated by the `import *`.

Reply:
The one below yes, but `CentralChi2Data` is only defined in multiclosure to my knowledge